x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/dataframe/evaluation/classification/MulticlassConfusionMatrix.java

```diff
@@ -142,7 +142,7 @@ public int getSize() {
         if (result.get() == null) {  // These are steps 2, 3, 4 etc.
             KeyedFilter[] keyedFiltersPredicted =
                 topActualClassNames.get().stream()
-                    .map(className -> new KeyedFilter(className, QueryBuilders.termQuery(predictedField, className)))
+                    .map(className -> new KeyedFilter(className, QueryBuilders.matchQuery(predictedField, className).lenient(true)))
```

Comment:
This will analyze the input string. Which might be ok, but it will mean if we have classNames `Foo` and `foo` we will treat them as the same.

Which, having two classes only differ by case is pathological and probably SHOULD result in error :)

Reply:
> This will analyze the input string. Which might be ok, but it will mean if we have classNames Foo and foo we will treat them as the same.

I added a test `testEvaluate_AllMetrics_KeywordField_CaseSensitivity` to verify that and it turns out casing is sensitive, i.e. "crocodile" does *not* match "cRoCoDiLe". This is because dependent variable can be of type `keyword` but not of type `text`.
I think comparing class labels in a case-sensitive manner is ok as long as we document it clearly. LMK if you think otherwise.

> Which, having two classes only differ by case is pathological and probably SHOULD result in error :)

Maybe or maybe not. E.g.: I could think of some scientific experiment results in which there would be class names named "a", "b", "c", "A", "B", "C" where "a" and "A" mean different things. I'm open to suggestions here as I don't have a strong opinion on that.

Reply:
WOOT! Ok, I did not think of the `keyword` mapping. Indeed, those are not analyzed. Scratch my previous concern.